app/src/main/java/com/techlads/composetv/features/home/navigation/topbar/HomeTopBar.kt

```diff
@@ -0,0 +1,152 @@
+
+    Column {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(24.dp)
+        ) {
+            TabRow(selectedTabIndex = selectedTabIndex,
+                indicator = { _, _ ->
+                    Box(
+                        modifier = Modifier
+                            .clip(MaterialTheme.shapes.small)
+                            .background(Color.Transparent)
+                    )
+                }
+            ) {
+                NavigationTabItem(item = MenuData.settingsItem,
+                    isSelected = selectedId == MenuData.settingsItem.id,
+                    onMenuSelected = {
+//                        selectedTabIndex = index
+                        onMenuSelected?.invoke(MenuData.settingsItem)
+                    })
+                MenuData.topBarMenuItems.forEachIndexed { index, menuItem ->
+                    NavigationTabItem(
+                        item = menuItem,
+                        isSelected = selectedId == menuItem.id,
+                        onMenuSelected = {
+                            selectedTabIndex = index
+                            onMenuSelected?.invoke(it)
+                        })
+                }
+            }
+            Text(
+                text = stringResource(id = R.string.app_name),
+                style = MaterialTheme.typography.bodyLarge.copy(fontWeight = FontWeight(700))
+            )
+        }
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+        ) {
+            content()
+        }
+    }
+}
+
+@OptIn(ExperimentalTvMaterial3Api::class)
+@Composable
+fun TabRowScope.NavigationTabItem(
+    item: MenuItem,
+    isSelected: Boolean,
+    enabled: Boolean = true,
+    onMenuSelected: ((menuItem: MenuItem) -> Unit)?
+) {
+    Tab(
+        selected = isSelected, enabled = enabled,
+        onFocus = {
+            onMenuSelected?.invoke(item)
+        },
+        colors = TabDefaults.pillIndicatorTabColors(
+            selectedContentColor = Color.Black,
+            focusedSelectedContentColor = if (item.id != NestedScreens.Settings.title) Color.Black else Color.White
+        ),
+        modifier = Modifier
+            .clip(if (item.id == NestedScreens.Settings.title) CircleShape else MaterialTheme.shapes.small)
+            .background(
+                if (isSelected) Color.White.copy(
+                    alpha = 0.7f
+                ) else Color.Transparent
+            )
+    ) {
+        when (item.id) {
+            NestedScreens.Search.title ->
+                Box(modifier = Modifier.size(36.dp), contentAlignment = Alignment.Center) {
+                    Icon(
+                        modifier = Modifier.size(24.dp),
+                        imageVector = item.icon ?: return@Tab,
```

Comment:
Never use a return statement in composable, I am also going to remove some of the places I have added in code, I learned that it generates unexpected results as it can disturb composition cycle

Reply:
Something like 

item.icon?.let {
     Icon(....)
}

Reply:
okay noted